### PR TITLE
pdctl: handle exceptions when PD is not running on the host

### PR DESCRIPTION
**Pull request by AstroProfundis**
Exceptions were not handled when querying PD APIs.

### CODE REVIEW

File: measurement/tidb/pdctl.py [hunk 1]
```diff
@@ -74,11 +74,17 @@ def build_url(uri):
...
-        fileutils.write_file(os.path.join(
-            full_outputdir, "%s-diagnose.json" % self.pd_host), self.read_diagnose())
+        pd_health = self.read_health()
+        if pd_health:
+            fileutils.write_file(os.path.join(
+                full_outputdir, "%s-health.json" % self.pd_host), pd_health)
+        pd_diagnose = self.read_diagnose()
+        if pd_diagnose:
+            fileutils.write_file(os.path.join(
+                full_outputdir, "%s-diagnose.json" % self.pd_host), self.read_diagnose())
```
Comment (ethercflow) [inline measurement/tidb/pdctl.py]: Is self.read_diagnose() => pd_diagnose better ?

Comment (AstroProfundis) [reply measurement/tidb/pdctl.py]: It's under class `PDCtl` so I think it's not necessary to label `pd_` again in method names.

Comment (ethercflow) [reply measurement/tidb/pdctl.py]: Sorry, I didn't express it clearly.  Well I mean `self.read_diagnose()` has been called at L81, so I think it's not necessary to call this method twice at L84.

Comment (AstroProfundis) [reply measurement/tidb/pdctl.py]: Ah, that was missed in the commit, just pushed a fix.

**Comment by AstroProfundis** [comment]
I think you should merge it.....

**Pull request merged**